test-app/app/src/main/assets/internal/ts_helpers.js

```diff
@@ -28,7 +28,34 @@
 
 	// For backward compatibility.
 	var __native = function(thiz) {
-		return thiz;
+	    // we are setting the __container__ property to the base class when the super method is called
+    	// if the constructor returns the __native(this) call we will use the old implementation
```

Comment:
Why do we need to support the old path where `__native(this)` copies properties?

Reply:
As there might be some plugins where they are using old typescript transpiled javascript code which won't get the result from the super method like this:
```Javascript
function MyButton10() {
    _super.call(this);
    this.myName = "MyName";
    return __native(this);
}
```
so the idea is for now to support both implementations and maybe in the future we will turn off the old one.

Reply:
Good! Any idea which TypeScript version introduced the new behavior?

Reply:
Nope